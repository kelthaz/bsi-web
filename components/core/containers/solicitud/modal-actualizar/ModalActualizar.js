import { useRouter } from 'next/router';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import Modal from '../../../../shared/modal/Modal';
import { resetChangePage, onHandleSubmit } from '../../../../../redux/actions/formulario';

const ModalActualizar = () => {
  const { push } = useRouter();
  const dispatch = useDispatch();
  const { routePage, showModal, isValid } = useSelector((state) => state.formulario);
  const [openModal, setOpenModal] = useState(false);

  const handleModalFirstOption = () => {
    setOpenModal(false);
    if (isValid) {
      dispatch(onHandleSubmit(true));
    } else {
      dispatch(resetChangePage());
    }
  };

  const handleModalSecondOption = async () => {
    setOpenModal(false);
    dispatch(resetChangePage());
    await push(routePage);
  };

  useEffect(() => {
    if (showModal) {
      setOpenModal(true);
    }
  }, [showModal]);

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal} onClose={() => dispatch(resetChangePage())}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
