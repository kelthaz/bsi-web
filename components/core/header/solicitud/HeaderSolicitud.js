import React from 'react';

const HeaderSolicitud = () => {
  return <div></div>;
};

export default HeaderSolicitud;
