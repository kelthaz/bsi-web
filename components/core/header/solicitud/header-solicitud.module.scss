@import '../../../../styles/scss/modules/var';
@import '../../../../styles/scss/modules/mixins';

.container-header {
  align-items: center;
  border-bottom: 1px solid $blue-shadow;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px 32px;

  @include media('sm') {
    padding: 14px 16px;
  }
}
