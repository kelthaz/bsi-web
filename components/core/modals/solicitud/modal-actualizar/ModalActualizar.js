import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import Modal from '../../../../shared/modal/Modal';

const ModalActualizar = ({ openModal, setOpenModal, formulario }) => {
  const { push } = useRouter();
  const { routePage } = useSelector((state) => state.formulario);

  const handleModalFirstOption = () => {
    if (formulario.isValid) {
      formulario.handleSubmit();
      push('/solicitud/[tab]/[step]', routePage);
    }
    setOpenModal(false);
  };

  const handleModalSecondOption = () => {
    push('/solicitud/[tab]/[step]', routePage);
    setOpenModal(false);
  };

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {formulario.isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {formulario.isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {formulario.isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
