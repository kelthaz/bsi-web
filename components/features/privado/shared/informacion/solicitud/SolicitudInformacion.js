import React from 'react';
import PropTypes from 'prop-types';
import { MORAL } from '../../../../../../constants/persona';

const SolicitudInformacion = ({ informacionSolicitud, tipoPersona }) => {
  const labels = {
    usoCredito: 'Destino del crédito',
    descripcionCredito: 'Razón social',
    domicilioFiscal: 'Domicilio fiscal',
    domicilioComercial: 'Domicilio comercial',
    nombreCompletoRecibe: 'Nombre de quien podría recibir token',
    celularRecibe: 'Teléfono de quien podría recibir token',
    telefonoEmpresa: 'Teléfono del negocio',
    curp: tipoPersona === MORAL ? 'CURP' : 'CURP del representante legal',
    estadoCivil: 'Estado civil del solicitante',
    nombreConyuge: 'Nombre de la Pareja',
    numeroEmpleados: 'Número de empleados',

    montoSolicitado: 'Crédito solicitado',
    montoAprobado: 'Crédito aprobado',

    tasaOrdinaria: 'Tasa ordinaria',
    tasaMoratoria: 'Tasa moratoria',
    comisionApertura: 'Comisión por apertura',
    plazo: 'Plazo del crédito',
    periodicidad: 'Esquema de pago',
    pago: 'Pagos',
    cat: 'CAT',
    tieneCuentaBancoppel: 'Tiene cuenta BanCoppel',
    clabe: 'Número de cuenta BanCoppel',
  };

  if (tipoPersona === MORAL) {
    delete labels.estadoCivil;
    delete labels.nombreConyuge;
  }

  const itemsPerfil = Object.keys(labels);

  return (
    <>
      {itemsPerfil.map((item) => (
        <div key={item} className="row pb-3">
          <div className="col-6">
            <div className="body2">{labels[item]}</div>
          </div>
          <div className="col-6">
            <div className="color-blue-storm body2">{informacionSolicitud[item]}</div>
          </div>
        </div>
      ))}
    </>
  );
};

SolicitudInformacion.propTypes = {
  informacionSolicitud: PropTypes.object.isRequired,
  tipoPersona: PropTypes.string.isRequired,
};

export default SolicitudInformacion;
