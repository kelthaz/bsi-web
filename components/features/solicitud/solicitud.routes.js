import cargaDocumentosRoutes from './carga-documentos/carga-documentos.routes';
import datosEmpresaRoutes from './datos-empresa/datos-empresa.routes';
import datosPersonalesRoutes from './datos-personales/datos-personales.routes';
import documentacionRoutes from './documentacion/documentacion.routes';
import ofertaRoutes from './oferta/oferta.routes';

const solicitudRoutes = [
  ...datosPersonalesRoutes,
  // ...datosEmpresaRoutes,
  // ...documentacionRoutes,
  // ...cargaDocumentosRoutes,
  // ...ofertaRoutes
].map((route) => ({ ...route, feature: 'solicitud' }));

export default solicitudRoutes;
