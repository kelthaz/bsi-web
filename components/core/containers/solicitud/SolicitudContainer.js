import { useRouter } from 'next/router';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { DATO_PERSONA } from '../../../../constants/formularios';
import { SIMULADOR_ROUTE } from '../../../../constants/routes/publico/publico';
import solicitudRoutes from '../../../features/solicitud/solicitud.routes';
import Step from '../../../shared/step/Step';
import TabInformativo from '../../../shared/tab-informativo/TabInformativo';
import SvgPatronesSolicitud from '../../../svgs/SvgPatronesSolicitud';
import HeaderSolicitud from '../../header/solicitud/HeaderSolicitud';
import ModalActualizar from '../../modals/solicitud/modal-actualizar/ModalActualizar';

const SolicitudContainer = ({ pageComponent, servicesData }) => {
  const { component, data } = pageComponent;
  const [componentPFAE, componentPM] = component;
  const { formulario, paso, step, tab, tipoPersona } = data;
  const { sincronizado, currentStep, tipoPersonaRedux } = useSelector((state) => state.solicitud);
  const { replace } = useRouter();

  let Component = <></>;

  if (tipoPersona === '' && componentPM) {
    Component = tipoPersonaRedux === 'MORAL' ? componentPM : componentPFAE;
  } else {
    Component = componentPFAE;
  }

  const componentsFormulario = solicitudRoutes
    .filter(({ data: dataComponent }) => dataComponent.formulario === formulario)
    .map(({ data: dataComponent, route }) => ({
      step: dataComponent.step,
      paso: dataComponent.paso,
      tipoPersona: dataComponent.tipoPersona,
      action: route,
    }));

  const steps = componentsFormulario.filter(
    (dataComponent) =>
      Number.isInteger(dataComponent.step) &&
      (dataComponent.tipoPersona === '' || dataComponent.tipoPersona === tipoPersonaRedux)
  );

  const tabs = [
    { path: 'datos-personales', label: 'Datos personales' },
    { path: 'datos-empresa', label: 'Datos de tu empresa' },
    { path: 'oferta', label: 'Oferta' },
    { path: 'documentacion', label: 'DocumentaciÃ³n' },
  ];

  useEffect(() => {
    if (!sincronizado && formulario === DATO_PERSONA) {
      replace(SIMULADOR_ROUTE);
    }
  }, []);

  return (
    <>
      <HeaderSolicitud />
      <ModalActualizar />

      {!!tab && (
        <TabInformativo
          tabs={tabs}
          currentTab={tab}
          currentStep={paso}
          valipStep={currentStep.paso}
          steps={componentsFormulario}
        />
      )}

      {!!step && <Step currentStep={step} valipStep={currentStep.valipStep} steps={steps} />}
      <Component {...servicesData} validate={currentStep.paso === paso} />
      <SvgPatronesSolicitud className="fixed-left-bottom" />
    </>
  );
};

SolicitudContainer.propTypes = {
  pageComponent: PropTypes.any.isRequired,
  servicesData: PropTypes.any.isRequired,
};

export default SolicitudContainer;
