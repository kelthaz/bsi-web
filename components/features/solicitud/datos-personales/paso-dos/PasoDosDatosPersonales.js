import { useFormik } from 'formik';
import { useDispatch, useSelector } from 'react-redux';
import * as Yup from 'yup';
import PropTypes from 'prop-types';
import { seleccionOpcion } from '../../../../../constants/errors';
import { PASO_TRES_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import useOnChangePage from '../../../../../hooks/useOnChangePage';
import { nextStepDatosPersonales } from '../../../../../redux/actions/solicitud';
import SvgPersonaFisicaActividadFisica from '../../../../svgs/SvgPersonaFisicaActividadFisica';
import SvgPersonaMoral from '../../../../svgs/SvgPersonaMoral';
import { FISICA, MORAL } from '../../../../../constants/persona';

const PasoDosDatosPersonales = ({ validate }) => {
  const { currentStep, datosPersonales } = useSelector((state) => state.solicitud);
  const dispatch = useDispatch();

  const formulario = useFormik({
    initialValues: {
      tipoPersona: datosPersonales.tipoPersona,
    },
    validationSchema: Yup.object().shape({
      tipoPersona: Yup.string().required(seleccionOpcion),
    }),
    onSubmit: (values) => {
      dispatch(
        nextStepDatosPersonales({
          currentStep: validate
            ? { ...currentStep, paso: currentStep.paso + 1, valipStep: currentStep.valipStep + 1 }
            : { ...currentStep },
          datosPersonales: {
            ...datosPersonales,
            ...values,
          },
        })
      );
    },
  });

  const { values, setFieldTouched, setFieldValue, touched } = formulario;

  const handleTipoPersona = async (tipoPersona) => {
    await setFieldValue('tipoPersona', tipoPersona);
    if (!touched.tipoPersona) {
      await setFieldTouched('tipoPersona', true);
    }
  };

  const [handleSubmit] = useOnChangePage(formulario, PASO_TRES_DATOS_PERSONA_ROUTE, validate);

  return (
    <div className="contedor-fixed-tab">
      <div className="contedor-solicitud">
        <div className="container p-0">
          <form onSubmit={handleSubmit} noValidate>
            <h2 className="color-blue-storm text-overflow">¡Hola, {datosPersonales.primerNombre}!</h2>
            <p className="color-dark-gray sub">Conozcámonos un poco más, ¿Qué tipo de persona eres?</p>

            <div className="row my-3">
              <div className="col-lg-6 col-md-6 col-sm-12 pl-lg-5 pl-md-5 col-xs-12 mb-sm-2 mb-xs-2">
                <button
                  type="button"
                  className={`card-simple-white-svg card-button ${
                    values.tipoPersona === FISICA && 'card-selected-blue-sky'
                  }`}
                  onClick={() => handleTipoPersona(FISICA)}
                >
                  <div>
                    <SvgPersonaFisicaActividadFisica />
                  </div>
                  <p className="px-md-5 px-lg-5">Persona Física con Actividad Empresarial</p>
                </button>
              </div>
              <div className="col-lg-6 col-md-6 col-sm-12 col-xs-12 pr-lg-5 pr-md-5">
                <button
                  type="button"
                  className={`card-simple-white-svg card-button ${
                    values.tipoPersona === MORAL && 'card-selected-blue-sky'
                  }`}
                  onClick={() => handleTipoPersona(MORAL)}
                >
                  <div>
                    <SvgPersonaMoral />
                  </div>

                  <p>Persona Moral</p>
                </button>
              </div>
            </div>
            <p className="color-gray-dark sub">
              <span>¿Eres persona física? ¡Adquiere tu crédito en </span>
              <a
                target="_blank"
                rel="noreferrer"
                href="https://www.bancoppel.com/ahorro_bcopp/cuenta_efectiva.html"
                className="link"
              >
                BanCoppel Personas!
              </a>
            </p>
            <div className="flex-column-center-config">
              <button
                type="submit"
                className="cicle-button-blue my-3"
                aria-label="Avanzar"
                disabled={validate && !(formulario.isValid && formulario.dirty)}
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

PasoDosDatosPersonales.propTypes = {
  validate: PropTypes.bool.isRequired,
};

export default PasoDosDatosPersonales;
