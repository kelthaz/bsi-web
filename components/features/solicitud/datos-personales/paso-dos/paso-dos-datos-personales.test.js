import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { useRouter } from 'next/router';
import { act } from 'react-dom/test-utils';
import storeTest from '../../../../../redux/storeTest';
import PasoDosDatosPersonales from './PasoDosDatosPersonales';
import { PASO_TRES_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import { FISICA, MORAL } from '../../../../../constants/persona';
import ModalActualizar from '../../../../core/containers/solicitud/modal-actualizar/ModalActualizar';

describe('Pruebas en el componente PasoDosDatosPersonales', () => {
  const store = storeTest();
  const push = jest.fn();
  const beforePopState = jest.fn();
  useRouter.mockImplementation(() => ({ push, beforePopState }));

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', async () => {
    const wrapper = mount(
      <Provider store={store}>
        <PasoDosDatosPersonales validate={false} />
        <ModalActualizar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test('Debe de guardar en el reducer el nuevo paso, tipo de persona FISICA y enviarlo al siguiente paso', async () => {
    // arrange
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoDosDatosPersonales validate />
      </Provider>
    );
    const form = wrapper.find('form');
    const personaFisica = wrapper.find('button').at(0);
    // act
    await act(async () => {
      personaFisica.simulate('click');
    });

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // assert
    expect(datosPersonales.tipoPersona).toEqual(FISICA);
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso + 1);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep + 1);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_TRES_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de guardar en el reducer el nuevo paso, tipo de persona MORAL y enviarlo al siguiente paso', async () => {
    // arrange
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoDosDatosPersonales validate />
      </Provider>
    );
    const form = wrapper.find('form');
    const personaMoral = wrapper.find('button').at(1);
    // act
    await act(async () => {
      personaMoral.simulate('click');
    });

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // assert
    expect(datosPersonales.tipoPersona).toEqual(MORAL);
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso + 1);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep + 1);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_TRES_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de mantener el paso anterior en el reducer y enviarlo al siguiente paso', async () => {
    // arrange
    const {
      solicitud: { currentStep },
    } = store.getState();

    const wrapper = mount(
      <Provider store={store}>
        <PasoDosDatosPersonales validate={false} />
        <ModalActualizar />
      </Provider>
    );

    const form = wrapper.find('form');
    const personaMoral = wrapper.find('button').at(1);
    const personaFisica = wrapper.find('button').at(0);
    // act
    await act(async () => {
      personaMoral.simulate('click');
    });
    await act(async () => {
      personaFisica.simulate('click');
    });

    await act(async () => {
      form.simulate('submit');
    });

    wrapper.update();

    await act(async () => {
      wrapper.find('.btn-medium-secondary').simulate('click');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();

    // assert
    expect(datosPersonales.tipoPersona).toEqual(FISICA);
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_TRES_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });
});
