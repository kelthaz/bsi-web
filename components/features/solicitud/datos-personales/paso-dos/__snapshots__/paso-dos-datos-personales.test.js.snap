// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PasoDosDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PasoDosDatosPersonales
    validate={false}
  >
    <div
      className="contedor-fixed-tab"
    >
      <div
        className="contedor-solicitud"
      >
        <div
          className="container p-0"
        >
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <h2
              className="color-blue-storm text-overflow"
            >
              ¡Hola, 
              JESUS
              !
            </h2>
            <p
              className="color-dark-gray sub"
            >
              Conozcámonos un poco más, ¿Qué tipo de persona eres?
            </p>
            <div
              className="row my-3"
            >
              <div
                className="col-lg-6 col-md-6 col-sm-12 pl-lg-5 pl-md-5 col-xs-12 mb-sm-2 mb-xs-2"
              >
                <button
                  className="card-simple-white-svg card-button false"
                  onClick={[Function]}
                  type="button"
                >
                  <div>
                    <SvgPersonaFisicaActividadFisica>
                      <svg
                        fill="none"
                        height={48}
                        viewBox="0 0 48 48"
                        width={48}
                      >
                        <path
                          d="M24.188 7.578a8.552 8.552 0 011.283 4.536c0 4.483-3.363 8.116-7.511 8.116-2.939 0-5.484-1.825-6.717-4.48"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                        <path
                          d="M11.243 15.75a8.611 8.611 0 01-.795-3.635C10.448 7.632 13.811 4 17.96 4c2.594 0 4.88 1.419 6.228 3.58"
                          stroke="#81C1EA"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                        <path
                          d="M27.386 27.142a23.424 23.424 0 00-17.697 0c-3.45 1.416-6.622 3.708-6.752 7.36 0 0-1.626 6.745 16.728 3.506"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                        <path
                          d="M32.384 31.257V27.94h6.946v3.316"
                          stroke="#81C1EA"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                        <path
                          d="M35.658 40.488c-.81 0-1.471-.662-1.471-1.472v-1.75c0-.809.662-1.47 1.471-1.47.81 0 1.472.661 1.472 1.47v1.75c0 .81-.662 1.472-1.472 1.472z"
                          fill="#81C1EA"
                        />
                        <path
                          d="M27.236 36.911h17.797"
                          stroke="#81C1EA"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                        <path
                          d="M42.746 44.682H28.971c-1.26 0-2.289-1.03-2.289-2.287v-8.13c0-1.258 1.03-2.29 2.29-2.29h13.774c1.258 0 2.29 1.03 2.29 2.29v8.127a2.298 2.298 0 01-2.29 2.29z"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.5}
                        />
                      </svg>
                    </SvgPersonaFisicaActividadFisica>
                  </div>
                  <p
                    className="px-md-5 px-lg-5"
                  >
                    Persona Física con Actividad Empresarial
                  </p>
                </button>
              </div>
              <div
                className="col-lg-6 col-md-6 col-sm-12 col-xs-12 pr-lg-5 pr-md-5"
              >
                <button
                  className="card-simple-white-svg card-button false"
                  onClick={[Function]}
                  type="button"
                >
                  <div>
                    <SvgPersonaMoral>
                      <svg
                        fill="none"
                        height={48}
                        viewBox="0 0 48 48"
                        width={48}
                      >
                        <path
                          d="M4 26.295V42.88c0 1.683 1.185 3.05 2.646 3.05h12.327"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.8}
                        />
                        <path
                          d="M23.893 26.192V6.052C23.893 4.367 22.708 3 21.247 3H6.646C5.185 3 4 4.367 4 6.052v20.243"
                          stroke="#81C1EA"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.8}
                        />
                        <path
                          d="M43.357 36.347v6.533c0 1.685-1.185 3.052-2.646 3.052H19.48c-1.461-.003-2.646-1.37-2.646-3.055v-7.45"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.8}
                        />
                        <path
                          d="M16.834 35.429v-6.185c0-1.686 1.185-3.052 2.646-3.052h21.234c1.46 0 2.645 1.366 2.645 3.052v7.103M7.85 10.155h5.134M7.85 15.582h5.134M7.85 21.01h5.134M7.85 26.439h5.134"
                          stroke="#81C1EA"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.8}
                        />
                        <path
                          d="M32.48 36.8h-3.916c-.926 0-1.679.866-1.679 1.937v7.192h7.272v-7.192c.003-1.07-.748-1.937-1.676-1.937z"
                          stroke="#225AA7"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeMiterlimit={10}
                          strokeWidth={2.8}
                        />
                      </svg>
                    </SvgPersonaMoral>
                  </div>
                  <p>
                    Persona Moral
                  </p>
                </button>
              </div>
            </div>
            <p
              className="color-gray-dark sub"
            >
              <span>
                ¿Eres persona física? ¡Adquiere tu crédito en 
              </span>
              <a
                className="link"
                href="https://www.bancoppel.com/ahorro_bcopp/cuenta_efectiva.html"
                rel="noreferrer"
                target="_blank"
              >
                BanCoppel Personas!
              </a>
            </p>
            <div
              className="flex-column-center-config"
            >
              <button
                aria-label="Avanzar"
                className="cicle-button-blue my-3"
                disabled={false}
                type="submit"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </PasoDosDatosPersonales>
  <ModalActualizar>
    <Modal
      closeable={true}
      onClose={[Function]}
      openModal={false}
      setOpenModal={[Function]}
    />
  </ModalActualizar>
</Provider>
`;
